src/snakemake/resources.py

```
@@ -502,104 +409,539 @@
+        return [
+            f"{name}={value.value}"
+            for name, value in self._data.items()
+            if not value.is_evaluable() and value.value is not None
+        ]
+
+    @classmethod
+    def decode_arg(cls, arg: str) -> Tuple[str, str]:
+        try:
+            return cast(Tuple[str, str], tuple(arg.split("=", maxsplit=1)))
+        except ValueError:
+            raise ValueError("Resources have to be defined as name=value pairs.")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)